velox/cider/VeloxToCiderExpr.cpp

```
@@ -225,18 +229,25 @@ std::shared_ptr<Analyzer::Expr> VeloxToCiderExprConverter::toCiderExpr(
   if (vExpr->name() == "gt" || vExpr->name() == "lt" ||
       vExpr->name() == "gte" || vExpr->name() == "lte" ||
       vExpr->name() == "eq" || vExpr->name() == "and" ||
-      vExpr->name() == "multiply") {
+      vExpr->name() == "multiply" || vExpr->name() == "plus" ||
+      vExpr->name() == "modulus") {
     auto type = getCiderType(vExpr->type(), false);
     auto inputs = vExpr->inputs();
     CHECK_EQ(inputs.size(), 2);
     SQLQualifier qualifier = SQLQualifier::kONE;
-    return std::make_shared<Analyzer::BinOper>(
-        type,
-        false,
-        getCiderSqlOps(vExpr->name()),
-        qualifier,
-        toCiderExpr(inputs[0], colInfo),
-        toCiderExpr(inputs[1], colInfo));
+    // TODO: bug here, need also check child expr is null or not
+    auto leftExpr = toCiderExpr(inputs[0], colInfo);
+    auto rightExpr = toCiderExpr(inputs[1], colInfo);
+    if (leftExpr && rightExpr) {
+      return std::make_shared<Analyzer::BinOper>(
+          type,
+          false,
+          getCiderSqlOps(vExpr->name()),
+          qualifier,
+          leftExpr,
+          rightExpr);
+    }
+    return nullptr;
```

should we throw exception instead of null pointer?

---

I returned the null pointer here and then check it in the Node translation loop. If we just throw exception when there is a case we can't handle anywhere, we will loose some meaning debug info such as what's the plannode related with this error.

---

It leads a problem to fallback design and its granularity. My understanding is that it should be op level as we always have the case not supported. If that's sth we didn't support, we should return and continue traversing the op tree. Your thoughts?